### PR TITLE
Add TechPreviewNoUpgrade Periodics for GCP and VSphere

### PR DESCRIPTION
**Pull request by JoelSpeed**
Following up on #19845 which prioritised the platforms being migrated to CCMs for 4.9, this adds GCP and VSphere as TechPreviewNoUpgrade release informing jobs.

For now I've included this in the 4.9 jobs but depending on when this merges, we may want to move straight to 4.10.

As of 4.10 GCP and VSphere will also have CCMs in TechPreview and as such this is important to ensure we have sufficient testing of the CCMs.

Note for vSphere, the periodics are scheduled staggered currently, by 15 minutes each. 0, 15, 30 and 45 past the hour. To avoid confusion with those, I've staggered these new ones to 2 hours earlier and on the 0th and 30th minute. This should ensure the suite has time to pass before the other periodics start and use the quota

### CODE REVIEW

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml [hunk 1]
```diff
@@ -183,6 +183,11 @@ tests:
   steps:
     cluster_profile: gcp-openshift-gce-devel-ci-2
     workflow: openshift-e2e-gcp
+- as: e2e-gcp-techpreview
+  interval: 48h
+  steps:
+    cluster_profile: gcp-openshift-gce-devel-ci-2
+    workflow: openshift-e2e-gcp-techpreview
```
Comment (petr-muller) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:190]: We added configs for 4.10, can you add these there as well?

- ci-operator/config/openshift/release/openshift-release-master__ci-4.10.yaml
- ci-operator/config/openshift/release/openshift-release-master__nightly-4.10.yaml

File: core-services/release-controller/_releases/release-ocp-4.9.json [hunk 2]
```diff
@@ -60,10 +60,18 @@
       "optional":true,
       "prowJob":{"name":"periodic-ci-openshift-release-master-nightly-4.9-e2e-gcp"}
     },
+    "gcp-techpreview":{
+      "optional":true,
+      "prowJob":{"name":"periodic-ci-openshift-release-master-ci-4.9-e2e-gcp-techpreview"}
+    },
```
Comment (petr-muller) [inline core-services/release-controller/_releases/release-ocp-4.9.json:66]: Also do the same in `core-services/release-controller/_releases/release-ocp-4.10.json`

**Comment by JoelSpeed** [comment]
@petr-muller I've updated this to just be 4.10, I think for now we are only really concerned for these platforms on 4.10. Will add to 4.9 at a later date if we decide that's necessary

**Comment by JoelSpeed** [comment]
The tests are failing for two reasons, one is https://github.com/openshift/origin/pull/26449, the other is a lack of a 4.10 nightly. Will try and run the rehearsals again once these two issues are resolved

**Comment by petr-muller** [comment]
@JoelSpeed I think it's fine to merge the jobs despite the failures. They will simply start giving signal once these preconditions start to hold

**Review by petr-muller — APPROVED** [review]

**Pull request merged**